doc/modules/neural_networks_supervised.rst

```diff
@@ -0,0 +1,357 @@
+:meth:`MLPClassifier.fit` for further information.
+
+.. topic:: Examples:
+
+ * :ref:`example_plot_mlp_alpha.py`
+ * :ref:`example_plot_mlp_nonlinear.py`
+
+
+Regression
+==========
+
+Class :class:`MLPRegressor` implements  
+a multi layer perceptron (MLP) that trains using backpropagation with no 
+activation function in the output layer. Therefore, it uses the square error as 
+the loss function, and the output is a set of continuous values.
+
+:class:`MLPRegressor` also supports multi-output regression, in 
+which a sample can have more than one target.
+
+
+Algorithms
+==========
+
+MLP trains using `Stochastic Gradient Descent
+<http://en.wikipedia.org/wiki/Stochastic_gradient_descent>`_, `Adam
+<http://arxiv.org/abs/1412.6980>`_, 
+or `L-BFGS <http://en.wikipedia.org/wiki/Limited-memory_BFGS>`_. 
+Stochastic Gradient Descent (SGD) computes the gradient of the loss function 
+with respect to a parameter that needs adaptation, i.e.
+
+.. math::
+
+    w \leftarrow w - \eta (\alpha \frac{\partial R(w)}{\partial w}
+    + \frac{\partial Loss}{\partial w})
+    
+where :math:`\eta` is the learning rate which controls the step-size in
+the parameter space search.  :math:`Loss` is the loss function used
+for the network.
+
+More details can be seen in the documentation of 
+`SGD <http://scikit-learn.org/stable/modules/sgd.html>`_ 
+
+Adam is similar to SGD in a sense that it's a stochastic
+optimization algorithm, but it can automatically adjust the amount to
+update parameters based on adaptive estimates of lower-order moments.
+
+With SGD or Adam, training supports online and mini-batch learning.
+
+L-BFGS is a fast learning algorithm that approximates the Hessian matrix which 
+is the second-order partial derivative of a function. Further it approximates 
+the inverse of the Hessian matrix to perform parameter update. 
+The implementation uses the Scipy version of 
+`L-BFGS <http://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.fmin_l_bfgs_b.html>`_..
+
+If the selected algorithm is 'L-BFGS', training does not support online nor 
+mini-batch learning.
+
+
+Complexity
+==========
+
+Suppose there are :math:`n` training samples, :math:`m` features, :math:`k` 
+hidden layers, each containing :math:`h` neurons - for simplicity, and :math:`o`
+output neurons.  The time complexity of backpropogation is 
+:math:`O(n\cdot m \cdot h^k \cdot o \cdot i)`, where :math:`i` is the number 
+of iterations. Since backpropagation has a high time complexity, it is advisable
+to start with smaller number of hidden neurons and few hidden layers for 
+training.
+
+
+Mathematical formulation
+========================
+
+Given a set of training examples :math:`(x_1, y_1), (x_2, y_2), \ldots, (x_n, y_n)` 
+where :math:`x_i \in \mathbf{R}^n` and :math:`y_i \in \{0, 1\}`, a one hidden 
+layer mlp learns the score function :math:`f(x) = W_2^T g(W_1^T x + b_1) + b_2` 
+where :math:`W_1, W_2 \in \mathbf{R}^m` and :math:`b_1, b_2 \in \mathbf{R}` are 
```

Comment:
If `W_1` is not a matrix, but a column vector, then `g` only outputs one number, so `W_2` would be a scalar.